rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,834 @@
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              examples:
+                NoSuchNamespaceExample:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+
+    post:
+      tags:
+        - Catalog API
+      summary: Create a namespace
+      description:
+        Create a namespace, with an optional set of properties.
+        The server might also add properties, such as last_modified_time etc.
+      operationId: createNamespace
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateNamespaceRequest'
+      responses:
+        200:
+          description: OK
+        401:
+          $ref: '#/components/responses/UnauthorizedResponse'
+        409:
+          description: Conflict - The namespace already exists
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              examples:
+                NamespaceAlreadyExists:
+                  $ref: '#/components/examples/NamespaceAlreadyExistsError'
+
+  /v1/namespaces/{namespace}:
+    parameters:
+      - $ref: '#/components/parameters/namespace'
+    get:
+      tags:
+        - Catalog API
+      summary: Load the metadata properties for a namespace
+      operationId: loadNamespaceMetadata
+      description: Return all stored metadata properties for a given namespace
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/GetNamespaceResponse'
+        401:
+          $ref: '#/components/responses/UnauthorizedResponse'
+        404:
+          description: Not Found - Namespace not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              examples:
+                NoSuchNamespaceExample:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a namespace exists
+      operationId: namespaceExists
+      description:
+        Check if a namespace exists.
+        This operation does not presently return a response body, which is why the response codes have only
+        a description field.
+      responses:
+        200:
+          description: Namespace exists
+        401:
+          description: Unauthorized
+        404:
+          description: Not Found
+
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a namespace from the catalog. Namespace must be empty.
+      operationId: dropNamespace
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/DropNamespaceResponse'
```

Comment:
Am I correct that the intent is to wrap all successful responses with `ResponseDataObject`? That doesn't appear to be what happens in this case because the schema of the response doesn't include the `data` layer.

I'm not sure if it would be very easy to adapt these schemas so that each one is wrapped by `ResponseDataObject`. Seems to me that would require some kind of inheritance, but I don't know how you'd do that with OpenAPI. If inheritance and customizing the `data` payload isn't possible, then we'd need to add that layer to every schema (which is already done for error examples, like `NoSuchNamespaceError`).

Modifying every schema doesn't seem worth it. I don't think that the `data` wrapper is very valuable and it would require a lot of additional boilerplate. The wrapper duplicates error codes anyway.

Another option is to add documentation somewhere that wrapping with `ResponseDataObject` is the convention, but I think it would be best if the actual definition matched.

Reply:
Yeah that was the intent, but I haven't settled on a good way to show that.

Some people have suggested getting rid of it, and having success responses not wrapped in a standard wrapper (which would be like it currently is). Which is why I've left it as is.

I'll count yours as the final vote against having the `data` wrapper and will remove that entirely. Success responses will just be the JSON object (like `DropNamespaceResponse`), but the error responses will all have the same standard error wrapper.

Reply:
Yeah, we should definitely remove it. If we can't express that the same wrapper should be used for all of these, then it doesn't make sense to have it. And the utility was dubious anyway.

Reply:
`ResponseDataWrapper` has been removed. There's now a schama `StandardErrorWrapper`, which has the `message`, `type`, `code` fields.

And then there's a response object, `IcebergErrorResponse` under `responses` which makes the response into `{ "error": { ... fields from error wrapper ... }`.

The success responses have no wrapper, and use their top level defined responses.